router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java

```diff
@@ -143,7 +143,7 @@ public DefaultRouteInfo(AnnotationMetadata annotationMetadata,
             this.consumesMediaTypes = consumesMediaTypes;
             consumesMediaTypesContainsAll = this.consumesMediaTypes.contains(MediaType.ALL_TYPE);
         }
-        this.imperative = isVoid || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
+        this.imperative = returnType.getType() == void.class || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
```

Comment:
Should we check the `isVoid` as well?

```suggestion
        this.imperative = (isVoid || returnType.getType() == void.class) || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
```

Reply:
No, that method includes async reactive methods

Reply:
*That field